core/src/main/java/com/yahoo/oak/NativeMemoryAllocator.java

```diff
@@ -196,13 +192,21 @@ public boolean isClosed() {
 
     // When some buffer need to be read from a random block
     @Override
-    public void readByteBuffer(Slice s) {
+    public void readByteBuffer(Slice s, int blockID) {
+
+        int sliceBlockID = s.getAllocatedBlockID(); // old block ID
+
+        // do we need to update the buffer?
+        if (sliceBlockID != INVALID_BLOCK_ID && sliceBlockID == blockID) {
+            return;
+        }
```

Comment:
We have other optimizations especially in the scan path the if we forgot about it and do something it can do something unexpected.
I don't see a bug with that if we expose the MM package.
Regarding whether or not it contributes to performance I am yet to see the performance results.